Delete databases created by previous `sqlite` tests so next test does not fail

### What should be done?

Our sqlite tests may fail with `SQL logic error: no such table: _ferretdb_collections (1)`. This is confusing, but it means there is an old database from previous test that is causing the test to fail.

I see that test database could be in `tmp/sqlite-tests/*.sqlite` if `task test-integration-sqlite` is run or unit tests in   `internal/backends/sqlite/metadata/*.sql` or `internal/backends/sqlite/metadata/pool/*.sql`. Wherever file database is used.

We could use `task reset-env` to ensure these databases are deleted. Or tests can ensure database is empty when it starts a test.

```
--- FAIL: TestDropIndexesCommandCompat (0.00s)
    --- FAIL: TestDropIndexesCommandCompat/DocumentIndex (0.01s)
        indexes_command_compat_test.go:385: 
                Error Trace:    /Users/elena/workdir/FerretDB/integration/setup/listener.go:184
                                                        /Users/elena/workdir/FerretDB/integration/setup/setup_compat.go:98
                                                        /Users/elena/workdir/FerretDB/integration/indexes_command_compat_test.go:396
                Error:          Received unexpected error:
                                [registry.go:85 metadata.NewRegistry] [registry.go:103 metadata.(*Registry).initCollections] SQL logic error: no such table: _ferretdb_collections (1)
                Test:           TestDropIndexesCommandCompat/DocumentIndex
        listener.go:155: Keeping /Users/elena/workdir/FerretDB/tmp/sqlite-tests/TestDropIndexesCommandCompat-DocumentIndex (file:../tmp/sqlite-tests/TestDropIndexesCommandCompat-DocumentIndex/) for debugging.
FAIL
```

See slack conversation too https://ferretdb.slack.com/archives/C02NWN1K3MJ/p1694593455832389


### Where?

https://github.com/FerretDB/FerretDB/tree/main/internal/backends/sqlite/metadata/pool
https://github.com/FerretDB/FerretDB/tree/main/internal/backends/sqlite/metadata
https://github.com/FerretDB/FerretDB/tree/main/integration

### Definition of Done

- unit tests added/updated;
- integration/compatibility tests added/updated;
- spot refactorings done;

--- a/integration/setup/listener.go
+++ b/integration/setup/listener.go
@@ -148,6 +148,7 @@ func setupListener(tb testtb.TB, ctx context.Context, logger *zap.Logger) string
 
 		dir, err := filepath.Abs(u.Opaque)
 		require.NoError(tb, err)
+		require.NoError(tb, os.RemoveAll(dir))
 		require.NoError(tb, os.MkdirAll(dir, 0o777))
 
 		tb.Cleanup(func() {

